query-planner-js/src/buildPlan.ts

```diff
@@ -983,22 +988,22 @@ class FetchGroup {
 
     const inputNodes = inputs ? inputs.toSelectionSetNode() : undefined;
 
-    const operation = this.isEntityFetch
+    let operation = this.isEntityFetch
       ? operationForEntitiesFetch(
           this.dependencyGraph.subgraphSchemas.get(this.subgraphName)!,
           this.selection,
           variableDefinitions,
-          fragments,
           operationName,
         )
       : operationForQueryFetch(
           this.rootKind,
           this.selection,
           variableDefinitions,
-          fragments,
           operationName,
         );
 
+    operation = operation.optimize(fragments);
```

Comment:
Note to reviewers: a very minor optimistic refactor. The `optimize` method was called in each of `operationForEntitiesFetch` and `operationForQueryFetch` which was both less DRY, but also less readable in the sense that we always want to call that method, and it has no reason to depend on the kind of operation we deal with.